src/scmdata/ops.py

```diff
@@ -557,45 +557,169 @@ def divide(self, other, op_cols, **kwargs):
     return type(self)(out)
 
 
-def integrate(self, out_var=None):
+def cumsum(self, out_var=None, check_annual=True):
     """
-    Integrate with respect to time
+    Integrate with respect to time using a cumulative sum
+
+    This method should be used when dealing with piecewise-constant timeseries (
+    such as annual emissions) or step functions. In the case of annual emissions,
+    each timestep represents a total flux over a whole year, rather than an
+    average value or point in time estimate. When integrating, one
+    can sum up each individual year to get the cumulative total, rather than using
+    an alternative method for numerical integration, such as the trapizoidal
+    rule which assumes that the values change linearly between timesteps.
+
+    This method requires data to be on uniform annual intervals.
+    :meth:`scmdata.run.ScmRun.resample` can be used to resample the data onto
+    annual timesteps.
+
+    The output timesteps are the same as the timesteps of the input, but since the
+    input timeseries are piecewise constant (i.e. a constant for a given year),
+    the output can be thought of as being a sum up to and including the last day
+    of a given year. The functionality to modify the output timesteps to an
+    arbitrary day/month of the year has not been implemented, if that would be
+    useful raise an issue on GitHub.
+
+    If the timeseries are piecewise-linear, :meth:`cumtrapz` should be used instead.
 
     Parameters
     ----------
     out_var : str
         If provided, the variable column of the output is set equal to
         ``out_var``. Otherwise, the output variables are equal to the input
-        variables, prefixed with "Cumulative " .
+        variables, prefixed with "Cumulative ".
+
+    check_annual : bool
+        If True (default), check that the timeseries are on uniform annual
+        intervals.
 
     Returns
     -------
     :class:`scmdata.ScmRun <scmdata.run.ScmRun>`
         :class:`scmdata.ScmRun <scmdata.run.ScmRun>` containing the integral of ``self`` with respect
         to time
 
+    See Also
+    --------
+    :meth:`cumtrapz`
+
+    Raises
+    ------
+    ValueError
+        If an unknown method is provided
+        Failed unit conversion
+        Non-annual timeseries and `check_annual` is False
+
     Warns
     -----
     UserWarning
         The data being integrated contains nans. If this happens, the output
         data will also contain nans.
+
     """
-    if not has_scipy:
-        raise ImportError("scipy is not installed. Run 'pip install scipy'")
+    time_unit = "a"
```

Comment:
This will go a bit pear-shaped if the user doesn't use `check_annual` right cause the assumed interval will be wrong?

Reply:
Yes, but let's fix that when we have a concrete use-case